android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt

```diff
@@ -120,18 +133,37 @@ class AnalyticsModule(private val context: ReactApplicationContext) : ReactConte
     @ReactMethod
     fun getCustomData(nativeId: NativeId, playerId: NativeId?, promise: Promise) {
         uiManager()?.addUIBlock { _ ->
-            collectors[nativeId]?.let {
-                promise.resolve(JsonConverter.fromAnalyticsCustomData(it.customData))
+            val source = playerModule()?.getPlayer(playerId)?.source
+            val collector = collectors[nativeId]
+            when {
+                source == null -> promise.reject(
+                    "[AnalyticsModule]", "Could not find source for player ($playerId)"
+                )
+                collector == null -> promise.reject(
+                    "[AnalyticsModule]", "Could not find analytics collector ($nativeId)"
+                )
+                else -> promise.resolve(JsonConverter.fromAnalyticsCustomData(collector.getCustomData(source)))
             }
         }
     }
 
     @ReactMethod
     fun addSourceMetadata(nativeId: NativeId, playerId: NativeId?, json: ReadableMap?) {
         uiManager()?.addUIBlock { _ ->
-            val playerSource = playerModule()?.getPlayer(playerId)?.source ?: return@addUIBlock
-            JsonConverter.toAnalyticsSourceMetadata(json)?.let { sourceMetadata ->
-                collectors[nativeId]?.addSourceMetadata(playerSource, sourceMetadata)
+            val source = playerModule()?.getPlayer(playerId)?.source
+            val collector = collectors[nativeId]
+            val sourceMetadata = JsonConverter.toAnalyticsSourceMetadata(json)
+            when {
+                source == null -> Log.d(
```

Comment:
Should these be warnings or errors?

Reply:
I kept them as debug, as this is mainly relevant for development and debugging. We could convert them, optimally you never see any of the messages anyway.